### DISCUSSION TITLE
How do I render a Quarto book more quickly?

### DISCUSSION THREAD

**Discussion by maelle** [quarto-dev/quarto-cli#6173] [Q&A]
### Description

When rendering a Quarto book, is there a way to update only the HTML files of the chapters whose source changed, even if that might lead to some outdated content elsewhere (cross-references, table of contents)? A quick partial render?

I first thought I was looking for caching/freezing but that might not be the case. I'm dealing with a book that is nearly only text. I'm rendering three versions of it (languages) then linking them through [babelquarto](https://docs.ropensci.org/babelquarto/). The use case of quick partial renders would be a local preview (whereas the book would be fully rendered on CI).

Any pointer would be appreciated. :smile_cat:

**Comment by maelle** [comment]
Now reading the docs of preview, but I'm not sure it's what I'm after. :thinking:

**Comment by andrewheiss** [comment]
I've run into similar issues with websites (https://github.com/quarto-dev/quarto-cli/discussions/3661) and the solution there is to use preview, since that can detect changed files and only render them, but I don't know how/if that works with (1) books and (2) multilanguage things like babelquarto :(

**Reply by maelle** [reply]
Thanks! Right, I fear preview wouldn't work well since I'm doing more than one render.

**Comment by hadley** [comment]
Is this for interactive writing? If you just Cmd + K to knit the current doc, it will only re-build the chapters that have changed.

**Reply by maelle** [reply]
Thanks! I'm slowly realizing my problem might be due to the exact setup of babelquarto, not Quarto (that solution works for the English version of my book :smile_cat:).
